monai/apps/deepgrow/dataset.py

```diff
@@ -0,0 +1,271 @@
+        transforms: explicit transforms to execute operations on input data.
+
+    Raises:
+        ValueError: When ``dimension`` is not one of [2, 3]
+        ValueError: When ``datalist`` is Empty
+
+    Returns:
+        A new datalist that contains path to the images/labels after pre-processing.
+
+    Example::
+
+        datalist = create_dataset(
+            datalist=[{'image': 'img1.nii', 'label': 'label1.nii'}],
+            base_dir=None,
+            output_dir=output_2d,
+            dimension=2,
+            image_key='image',
+            label_key='label',
+            pixdim=(1.0, 1.0),
+            limit=0,
+            relative_path=True
+        )
+
+        print(datalist[0]["image"], datalist[0]["label"])
+    """
+
+    if dimension not in [2, 3]:
+        raise ValueError("Dimension can be only 2 or 3 as Deepgrow supports only 2D/3D Training")
+
+    if not len(datalist):
+        raise ValueError("Input datalist is empty")
+
+    transforms = _default_transforms(image_key, label_key, pixdim) if transforms is None else transforms
+    new_datalist = []
+    for idx in range(len(datalist)):
+        if limit and idx >= limit:
+            break
+
+        image = datalist[idx][image_key]
+        label = datalist[idx].get(label_key, None)
+        if base_dir:
+            image = os.path.join(base_dir, image)
+            label = os.path.join(base_dir, label) if label else None
+
+        image = os.path.abspath(image)
+        label = os.path.abspath(label) if label else None
+
+        logging.info("Image: {}; Label: {}".format(image, label if label else None))
+        data = transforms({image_key: image, label_key: label})
+        if dimension == 2:
+            data = _save_data_2d(
+                vol_idx=idx,
+                vol_image=data[image_key],
+                vol_label=data[label_key],
+                dataset_dir=output_dir,
+                relative_path=relative_path,
+            )
+        else:
+            data = _save_data_3d(
+                vol_idx=idx,
+                vol_image=data[image_key],
+                vol_label=data[label_key],
+                dataset_dir=output_dir,
+                relative_path=relative_path,
+            )
+        new_datalist.extend(data)
+    return new_datalist
+
+
+def _default_transforms(image_key, label_key, pixdim):
+    keys = [image_key] if label_key is None else [image_key, label_key]
+    mode = [GridSampleMode.BILINEAR, GridSampleMode.NEAREST] if len(keys) == 2 else [GridSampleMode.BILINEAR]
+    return Compose(
+        [
+            LoadImaged(keys=keys),
+            AsChannelFirstd(keys=keys),
+            Spacingd(keys=keys, pixdim=pixdim, mode=mode),
+            Orientationd(keys=keys, axcodes="RAS"),
+        ]
+    )
+
+
+def _save_data_2d(vol_idx, vol_image, vol_label, dataset_dir, relative_path):
+    data_list = []
+
+    if len(vol_image.shape) == 4:
```

Comment:
If the input is 3D NIFTI images with shape [H, W, D]
After the default pre-transforms
```        
        [
            LoadImaged(keys=keys),
            AsChannelFirstd(keys=keys),
            Spacingd(keys=keys, pixdim=pixdim, mode=mode),
            Orientationd(keys=keys, axcodes="RAS"),
        ]
 ```
 it will be [D, H, W]
 
 If the input is 4D NIFTI images with shape [H, W, D, C]
 After the default pre-transforms it will be [C, H, W, D]
 So that's why this condition is added here
 To get just the first channel and make it to be [D, H, W]
 
 This is what I observe.
 Sachi can correct me if I am wrong

Reply:
thanks, if that's the case, then the current line 216 in `_save_data_3d` is incorrect?

Reply:
I think line 216 is correct.

For save 2d and save 3d, that means saving the data for either DeepGrow 2D training or DeepGrow 3D training.
For DeepGrow 2D training we will save the image to [H, W] pictures.
While for DeepGrow 3D training we save the image to [D, H, W] volume.